src/torManager/torManager.test.ts

```diff
@@ -98,4 +98,55 @@ describe('Tor manager', () => {
     expect(tor.torHashedPassword).toHaveLength(61)
     expect(tor.torPassword).toHaveLength(32)
   })
+
+  it('tor spawn repeating 3 times with 1 second timeout and repeating will stop after that', async () => {
+    const torPath = torBinForPlatform()
+    const [controlPort] = await fp(9051)
+    const httpTunnelPort = (await fp(controlPort as number + 1)).shift()
+    const socksPort = (await fp(httpTunnelPort as number + 1)).shift()
+    const libPath = torDirForPlatform()
+    const tor = new Tor({
+      appDataPath: tmpAppDataPath,
+      socksPort,
+      torPath: torPath,
+      controlPort,
+      httpTunnelPort,
+      options: {
+        env: {
+          LD_LIBRARY_PATH: libPath,
+          HOME: tmpAppDataPath
+        },
+        detached: true
+      }
+    })
+
+    try {
+      await tor.init(3, 1000)
+      await tor.kill()
+    } catch { console.log('to many try of tor spawn') }
```

Comment:
I understand that this test case expects to get to this catch, right?
Correct way to test for it is something like:
```
const failToStartTorBecauseOfTimeout = async () {
      await tor.init(3, 1000)
}
expect(failToStartTorBecauseOfTimeout).toThrow("Failed to spawn tor 3 times");

await tor.kill()
```